spatial/r3/mat.go

```diff
@@ -0,0 +1,235 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import (
+	"gonum.org/v1/gonum/blas/blas64"
+	"gonum.org/v1/gonum/mat"
+)
+
+const (
+	badDim = "bad matrix dimensions"
+	badIdx = "bad matrix index"
+)
+
+// Mat represents a 3×3 matrix. Useful for rotation matrices and such.
+type Mat struct {
+	data *[3][3]float64
+}
+
+var _ mat.Matrix = (*Mat)(nil)
```

Comment:
I'm wondering about whether we need to have a pointer on `Mat`'s methods. Formally we don't since the mutations are to the `data` field, but it makes it look like a `Mat` is immutable. I'm think I'm still leaning towards removing it because it reduces the amount of pointer walking required. It would be helpful to see the difference in the generated code between the two.

@sbinet What do you think?

Reply:
You do need a pointer receiver if you want to re-assign the backing slice if I'm not mistaken, as we are doing in `setBackingSlice`

Reply:
We can leave it as it is.